app/models/user.rb

```diff
@@ -1,9 +1,23 @@
+require "securerandom"
+
 class User < ApplicationRecord
 
   has_secure_password
+  has_secure_password :reset_token, validations: false
   validates_confirmation_of :password
-  validates :password, length: { minimum: 8 }
+  validates :password, length: { minimum: 8 }, unless: :skip_password_verification
   validates :email, presence: true, email: true, uniqueness: true
   has_one :person, dependent: :destroy
 
+  attr_accessor :skip_password_verification
+
+  def generate_reset_token
+    token = SecureRandom.hex(20)
```

Comment:
This generates a 20-character random hex (containing characters from 0-9 and A-F) string.